Tribes/src/app/building.service.ts

```diff
@@ -20,4 +24,12 @@ export class BuildingService {
   getSoldiersFromAPI(): Observable<SoldierResponse> {
     return this.http.get<SoldierResponse>(ROOT_URL + '/kingdom/soldiers');
   }
+
+  addNewBuilding(type: string): void {
+    this.http.post(ROOT_URL + '/kingdom/buildings', {type})
+    .pipe(catchError(this.errorHandlingService.handleError))
+    .subscribe(response => { this.beginConstruction.emit(response);
+                             setTimeout(() => { this.finishConstruction.emit(); }, 1000 * CONSTRUCTION_TIME); },
```

Comment:
You could place this callback to a separate function, and reference it by name here